presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -866,7 +866,7 @@ else if (equality.getRight().equals(column) && columns.contains(equality.getLeft
                         .filter(SymbolReference.class::isInstance)
                         .map(Symbol::from)
                         .collect(toImmutableSet());
-                if (symbolsInAssignment.containsAll(symbolInExpression)) {
+                if (symbolsInAssignment.equals(symbolInExpression)) {
```

Comment:
This is not enough. It needs to make sure the arguments to both coalesce expressions are *only* simple symbol references. Here are a couple of queries that don't produce the right results:

```sql
SELECT coalesce(t.a, s.a + 1 - 1, r.a)
FROM (VALUES 1, 2, 3) t(a)
FULL OUTER JOIN (VALUES 1, 4) s(a) ON t.a = s.a
FULL OUTER JOIN (VALUES 2, 5) r(a) ON t.a = r.a
GROUP BY 1
```

```sql
SELECT coalesce(t.a, u.a, if(t.b is null, 100, t.b)), count(*)
FROM (VALUES (1, 10), (2, 20), (3, 30), (null, 40), (100, 50)) t(a, b)
FULL OUTER JOIN (VALUES 1, 4, null) u(a) ON t.a = u.a
GROUP BY 1;
```